Adds Nginx Integrator Charms in the bundle

claudiubelu opened the pull request:
Removes the ``finos-`` prefixes and ``-k8s`` suffixes from the application names, shortening them.

Adds an ``nginx-ingress-integrator`` charm for each of the following Legend charms: SDLC, Engine, Studio, and relates them.

claudiubelu committed:
Adds Nginx Integrator Charms in the bundle

Adds an nginx-ingress-integrator charm for each of the following
Legend charms: SDLC, Engine, Studio, and relates them.

pedroleaoc commented on bundle.yaml:
```diff
@@ -46,6 +61,12 @@ relations:
   - ["finos-legend-engine-k8s", "finos-legend-gitlab-integrator-k8s"]
   - ["finos-legend-studio-k8s", "finos-legend-gitlab-integrator-k8s"]
+  # Gitlab relations:
+  - ["finos-legend-gitlab-integrator-k8s", "gitlab-k8s"]
```
What is gitlab-k8s?

claudiubelu replied:
Was working on something in parallel, I forgot to remove this one. Good catch. :)

jnsgruk commented on bundle.yaml:
```diff
@@ -36,6 +36,21 @@ applications:
+  nginx-ingress-integrator-sdlc:
+    charm: "nginx-ingress-integrator"
+    channel: "stable"
+    scale: 1
```
I think we need a `trust` attribute here? (and on the others...)
```suggestion
    scale: 1
    trust: true
```

claudiubelu replied:
Done

jnsgruk commented on bundle.yaml:
```diff
@@ -36,6 +36,21 @@ applications:
+  nginx-ingress-integrator-engine:
+    charm: "nginx-ingress-integrator"
+    channel: "stable"
+    scale: 1
```
```suggestion
    scale: 1
    trust: true
```

claudiubelu replied:
Done

jnsgruk commented on bundle.yaml:
```diff
@@ -36,6 +36,21 @@ applications:
+  nginx-ingress-integrator-studio:
+    charm: "nginx-ingress-integrator"
+    channel: "stable"
+    scale: 1
```
```suggestion
    scale: 1
    trust: true
```

claudiubelu replied:
Done

jnsgruk commented on bundle.yaml:
```diff
@@ -36,6 +36,21 @@ applications:
     channel: "stable"
     scale: 1
 
+  nginx-ingress-integrator-sdlc:
```
```suggestion
  -legend-sdlc-ingress:
```

We could consider some shorter names here?

jnsgruk replied:
On this, we could rename a bunch of these charms (just in the bundle.yaml) so you get:

- mongodb
- legend-studio
- legend-sdlc
- legend-engine
- gitlab-integrator
- legend-db

Just makes typing juju commands a bit shorter! Thoughts @pedroleaoc / @balbirthomas ?

balbirthomas replied:
- Sounds good to me but we will need to change all the charms also (right ?). 
- What about "k8s" suffix (since these are k8s charms) ?

jnsgruk replied:
No, changing the name at the top level in the bundle is just like deploying a charm with an alias: `juju deploy mongodb-k8s mongodb`, for example

claudiubelu replied:
Done

claudiubelu committed:
Shortens the bundle application names

Removes the finos- prefixes and -k8s suffixes from the application
names, shortening them.

claudiubelu reviewed (commented):
Also renamed the deployed application names.
